app/src/main/java/ca/bc/gov/bchealth/ui/home/manage/QuickAccessManagementScreen.kt

```diff
@@ -0,0 +1,189 @@
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import ca.bc.gov.bchealth.R
+import ca.bc.gov.bchealth.compose.BasePreview
+import ca.bc.gov.bchealth.compose.MyHealthTheme
+import ca.bc.gov.bchealth.compose.MyHealthTypography
+import ca.bc.gov.bchealth.compose.bold
+import ca.bc.gov.bchealth.compose.theme.primaryBlue
+import ca.bc.gov.bchealth.compose.theme.statusBlue
+import ca.bc.gov.bchealth.compose.theme.white
+import ca.bc.gov.bchealth.ui.home.QuickAccessTileItem
+
+@Composable
+fun QuickAccessManagementScreen(
+    viewModel: QuickAccessManagementViewModel,
+    onClickItem: (QuickAccessTileItem) -> Unit,
+    onUpdateCompleted: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val uiState = viewModel.uiState.collectAsState().value
+
+    LaunchedEffect(Unit) {
+        viewModel.loadTilesUi()
+    }
+
+    QuickAccessManagementContent(uiState, onClickItem, modifier)
+
+    if (uiState.isUpdateCompleted) {
+        onUpdateCompleted.invoke()
+    }
+}
+
+@Composable
+private fun QuickAccessManagementContent(
+    uiState: QuickAccessManagementViewModel.QuickAccessManagementUiState,
+    onClickItem: (QuickAccessTileItem) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Box(modifier = modifier) {
+        LazyColumn(
+            modifier = modifier,
+            contentPadding = PaddingValues(start = 32.dp, end = 32.dp, top = 20.dp, bottom = 64.dp),
+            content = {
+                item {
+                    Text(
+                        text = stringResource(id = R.string.quick_access_management_body),
+                        style = MyHealthTypography.body1,
+                        color = primaryBlue
+                    )
+                }
+
+                item { Spacer(modifier = Modifier.size(16.dp)) }
+
+                for ((category, tiles) in uiState.uiMap) {
+                    item {
+                        Text(
+                            text = stringResource(id = category),
+                            style = MyHealthTypography.body1.bold(),
+                            color = statusBlue
+                        )
+                    }
+
+                    item { Spacer(modifier = Modifier.size(12.dp)) }
+
+                    items(tiles) { tile ->
+                        TileItemUi(tile, onClickItem)
+                        Spacer(modifier = Modifier.size(10.dp))
+                    }
+
+                    item { Spacer(modifier = Modifier.size(6.dp)) }
+                }
+            }
+        )
+
+        if (uiState.isLoading) {
```

Comment:
Should lift this thing out of the content option and should be handled inside of the screen method block.

Reply:
done